src/Drush/Commands/HooksDrushCommands.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Acquia\Drupal\RecommendedSettings\Drush\Commands;
+
+use Acquia\Drupal\RecommendedSettings\Helpers\EnvironmentDetector;
+use Consolidation\AnnotatedCommand\CommandData;
+use Consolidation\AnnotatedCommand\Events\CustomEventAwareInterface;
+use Consolidation\AnnotatedCommand\Events\CustomEventAwareTrait;
+use Consolidation\AnnotatedCommand\Hooks\HookManager;
+use Drush\Attributes as CLI;
+use Drush\Commands\DrushCommands;
+
+/**
+ * This
+ */
+class HooksDrushCommands extends DrushCommands {
+
+  /**
+   * Generate settings.php for multisite if current env. is not Site factory.
+   */
+  #[CLI\Hook(type: HookManager::ON_EVENT, target: MultisiteDrushCommands::VALIDATE_GENERATE_SETTINGS)]
+  public function isAcquiaEnvironment(): bool {
+    return !EnvironmentDetector::isAcsfEnv();
+  }
+
+  /**
+   * Generate settings.php for multisite if current env. is local or CloudIDE.
+   */
+  #[CLI\Hook(type: HookManager::ON_EVENT, target: MultisiteDrushCommands::VALIDATE_GENERATE_SETTINGS)]
+  public function isLocalEnvironment(): bool {
+    return (EnvironmentDetector::isLocalEnv() || EnvironmentDetector::isAhIdeEnv());
+  }
```

Comment:
I think we should use single method invocation instead of multiple, it utilises the process thread. Combine the above two method invocation in one like below:
```
/**
   * Generate settings.php for multisite if current env. is local or CloudIDE but not ACSF.
   */
  #[CLI\Hook(type: HookManager::ON_EVENT, target: MultisiteDrushCommands::VALIDATE_GENERATE_SETTINGS)]
  public function isLocalEnvironment(): bool {
    return (!EnvironmentDetector::isAcsfEnv() && (EnvironmentDetector::isLocalEnv() || EnvironmentDetector::isAhIdeEnv()));
  }
```

Reply:
Done.